Emscripten-build-instructions.md

```diff
@@ -0,0 +1,159 @@
+# Wasm Build Instructions
+
+It should be noted that the wasm build of CppInterOp is still experimental and subject to change to change.  
+
+## CppInterOp Wasm Build Instructions
+
+This document first starts with the instructions on how to build a wasm build of CppInterOp. Before we start it should be noted that  
+unlike the non wasm version of CppInterOp we currently only support the Clang-REPL backend using llvm>19 for osx and Linux.  
+We will first make folder to build our wasm build of CppInterOp. This can be done by executing the following command
+
+```bash
+mkdir CppInterOp-wasm
+```
+
+Now move into this directory using the following command
+
+```bash
+cd ./CppInterOp-wasm
+```
+
+To create a wasm build of CppInterOp we make use of the emsdk toolchain. This can be installed by executing (we only currently  
+support version 3.1.45)
+```bash
+git clone https://github.com/emscripten-core/emsdk.git
+cd emsdk
+./emsdk install  3.1.45
+```
+
+Go back upto the top level build directory using
+
+```bash
+cd ../..
+```
+
+and activate the emsdk environment
+
+```bash
+./emsdk/emsdk activate 3.1.45
+source ./emsdk/emsdk_env.sh
+```
+
+Now clone the 19.x release of the LLVM project repository and CppInterOp (the building of the emscripten version of llvm can be
+avoided by executing micromamba install llvm -c <https://repo.mamba.pm/emscripten-forge> and setting the LLVM_BUILD_DIR appropriately)
+
+
+```bash
+git clone --depth=1 --branch release/19.x https://github.com/llvm/llvm-project.git
+git clone --depth=1 https://github.com/compiler-research/CppInterOp.git
+```
+
+Now move into the cloned llvm-project folder and apply a required patch
+
+```bash
+cd ./llvm-project/
+git apply -v ../CppInterOp/patches/llvm/emscripten-clang${{ matrix.clang-runtime }}-*.patch
+```
+
+We are now in a position to build an emscripten build of llvm by executing the following
+```bash
+mkdir build
+cd build
+emcmake cmake -DCMAKE_BUILD_TYPE=Release \
+                        -DLLVM_HOST_TRIPLE=wasm32-unknown-emscripten \
+                        -DLLVM_ENABLE_ASSERTIONS=ON                        \
+                        -DLLVM_TARGETS_TO_BUILD="WebAssembly" \
+                        -DLLVM_ENABLE_LIBEDIT=OFF \
+                        -DLLVM_ENABLE_PROJECTS="clang;lld" \
+                        -DLLVM_ENABLE_ZSTD=OFF \
+                        -DLLVM_ENABLE_LIBXML2=OFF \
+                        -DCLANG_ENABLE_STATIC_ANALYZER=OFF \
+                        -DCLANG_ENABLE_ARCMT=OFF \
+                        -DCLANG_ENABLE_BOOTSTRAP=OFF \
+                        -DCMAKE_CXX_FLAGS="-Dwait4=__syscall_wait4" \
+                        ../llvm
+emmake make clang -j $(nproc --all)
+emmake make clang-repl -j $(nproc --all)
+emmake make lld -j $(nproc --all)
```

Comment:
I am not fully convinced here (not sure it matters in the build) but 
1) if we use `LLVM_ENABLE_PROJECTS="clang;lld"` that means we want to build all static libraries in this cases coming out of clang and lld
2) Our focus is not to get clang-repl (the approach we use to get clang-repl working in the browser doesn't need `clang-repl.js` or `clang-repl.wasm` rather everything works through `libclangInterpreter.a`)
3) Even if we want to build clang-repl (which has dependency on lldwasm and lldcommon, I guess this would invoke "make lld" already, as required to get clang-repl ..... which in turn makes the below command redundant.

Basically what I am saying is we need a simple `emmake make -j $(nproc --all)` here and nothing else. I had raised a similar point for the CI not sure if that was addressed.

Reply:
If we just need `libclangInterpreter.a` we should call that -- it should be a target : `make clangInterpreter` or something like that...

Reply:
Ahhh sorry I meant, we need libclangIntepreter.a (and all its respective transitive dependencies like lldwasm, lldcommon from lld ,  clangast, clangbasic etc from clang and llvmsupport etc from llvm)

I think just doing `emmake make clang -j $(nproc --all)` would get the job done

Reply:
I think this is fixed now. I removed the clang-repl target, leaving just the clang and lld ones.